graders/c/cgrader/check.h

```diff
@@ -73,10 +78,17 @@ static inline void pl_fixture_sandbox_setup(void) {
   // Clear all environment variables, including those used to configure libcheck
   clearenv();
 #endif
+
+  if (*sanitizer_output) {
+    mkstemp(sanitizer_output);
```

Comment:
Doesn't the `template` argument to `mkstemp` need to include `XXXXXX` as the last six characters? AFAICT `sanitizer_output` is the empty string.

Reply:
This function is called if `*sanitizer_output` has been initialized in `pl_setup_asan_hooks`, in which case this string will have the required XXXXXX. The idea is that this call is only done if ASAN hooks are set up, and the value of `*sanitizer_output` is a way to check if that is the case.

I originally tried to initialize the temp directory there, but the call to `closefrom` a few lines above interferes with that.

Reply:
Technically because `*sanitizer_output == '\0'`, but yes, that's it.